autogpt_platform/backend/backend/notifications/notifications.py

```
@@ -722,12 +726,65 @@ async def _process_batch(self, message: str) -> bool:
                             i += len(chunk)
                             chunk_sent = True
                             break
+                    except PostmarkException as e:
+                        if attempt_size == 1:
+                            # Single notification failed - determine the actual cause
+                            error_code = getattr(e, "error_code", None)
```

Ruff (0.13.2)

Local variable `error_code` is assigned to but never used

Remove assignment to unused variable `error_code`

(F841)